### DISCUSSION TITLE
Least common multipliers of frequencies. Do they exist?

### DISCUSSION THREAD

**Discussion by jimishol** [thetwom/Tuner#80] [General]
I am trying to understand some harmony aspects even if I luck any previous descent knowledge on the subject. I downloaded “tuner” app to try some experiment. ( I cannot but to congratulate developer for this app).

I understand what I read in wikipedia about [multiphonic tones](https://en.wikipedia.org/wiki/Multiphonic) that mostly refer to the sum and difference of 2 tones. Indeed by
cos(2*π*a*t) + cos(2*π*b*t) = 2*cos(2*π*(a+b)/2*t)*cos(2*π*(a-b)/2*t),
when a is almost equal to b, we see why multiphonic tones are relevant to the average of a and b frequencies, accompanied by a variant of volume, because when frequency (a-b)/2<20Hz it has long period and can not be heard as frequency. (The actually heard [beat frequency](https://en.wikipedia.org/wiki/Beat_(acoustics)) a-b is irrelevant to my question.)

My question is about the least common multiplier (lcm) of the periods of any sum of frequencies. When the ratio between frequencies is simple rational numbers, there is a pattern that is repeated in every lcm period.
I assumed that, even if in equally tempered tunes the pitches are not related by rational numbers at all, if the ratios are approximated well enough by simple rational numbers the lcm period, at least briefly, should still exist.
 
My question is if the mentioned above 1/lcm frequency is actually heard or is it just a mathematical, artificial frequency.

By the help of [maxima CAS](https://maxima.sourceforge.io/) I created the below function that calculates that 1/lcm frequency.

I downloaded the “tuner” app just to check this. So, in [musescore](https://musescore.org/en) I played E5+G5 and was very happy to see that the app picked up the right C3 pitch.
F(4,7) gives [¼, [4,7],0] – The result is that, we gave the [4=E and 7=G] tones and these produce a pattern that repeats in the ¼ of the frequency of the temperament tone on C, that correspond on C itself.
Explanation: E=5/4 of C, G=3/2 of C with periods 4/5=12/15 and 10/15 of C. The lcm of these are 60/15=4, so the frequency of the repeated pattern is ¼.

I am in question if that C3 picked by the “tuner” app is really heard.
(If it is heard, then iii chord is more close to I7 with omitted root than to the dominant or predominant regions.)
Unfortunately, considering the (a) minor scale, it is not true for all cases. #C+E, that is F(1,4), do not produce the expected 9=A but 0=C. Neither the “tuner” app pick the A3 pitch. The fact that A4=440Hz, as reference frequency, is seen down left of screen was misleading to me. Since C3 from F(4,7) was picked rightly, I was expected to see C4 as reference frequency instead of A4. The reason is that in E+G we combine 5/4 and 3/2 rationals. In the case of #C+E, we combine 17/16 and 5/4 rationals that produce different lcm.

Of course if I define my constant C as 3, then #C+E as F(1,4) indeed results on A. The “tuner” apps seems as it is picked the right A pitch only if I choose Temperament as A pitch in options. But if this option affects the picked up pitch, I wonder if these picked up tones are real or not.
An option would be to accept as Temperament tone always the first of arguments and resolve the results as the difference from there. In that case my relevant function would be
```
G([arguments]):= block([i,x,y,marg,minmarg,sol,ratprint,ratepsilon],
    ratprint:false,
    ratepsilon:1e-2,
marg:mod(arguments,12),
    minmarg:first(marg),
    marg:mod(marg-minmarg,12),
y:1/lcm(rat(bfloat(2^(-args(marg)/12)))),
i:0,
    while y*2^i < 1 do i:i+1,
sol:round(rhs(solve([2^(x/12)=2^i*y],[x])[1])),
[y,arguments,mod(sol+minmarg,12)]
)$
```

In summary I ask
if, in theory, the lcm pitches, picked up by the “tuner” app, from played chords (sum of tones) are really heard even unconsciously and, for you,
to consider if it would be more right the Temperament to be visual on screen instead of Reference frequency or even if they should be coincide by default, before any editing on them. 

Congratulations for the app and thank you.

**Comment by thetwom** [comment]
Hi,
 this is a long question and it is a bit hard to follow what your actually question is. 

I have the feeling that you are mixing things a bit up. Note names like C, D, E have not much to do about actual frequencies. It just gives names to certain frequencies. Although you can play notes like the mentioned E and F and try to compute ratios, it still depends on the temperament if you get pure ratios or not. You could even define your own temperament and say C4 = 100Hz, D4=101Hz, ... B4=110Hz and then C5 = 200Hz, and while  chords would not sound nice at all, it you can still do it :-).  So forget a second about note names.

In order to detect a tone, you can indeed try to find the least common multiplier. So, if you find 200Hz and 300Hz in a signal, this would be 100Hz as the base frequency (or least common multiplier as you name it). Would you hear 100Hz? Very often yes, but I bet that it depends. If the 200Hz is much, much quieter than the 300Hz, you might hear 300Hz instead. So you have to take this into account. There are algorithms which make it easier to find the base frequency, also taking into account the amplitudes. The tuner app uses autocorrelation for this.

So one step of the tuner app is to find this "least common multiplier". This is simply a frequency. It does not depend on temperament, reference frequency and so on. 

About the reference tone: This is only important AFTER you got the base frequency. Lets say, you found 220Hz. Then you "simply" look up which note is 440Hz. Here is where the reference tone comes into play. If you tell the tuner, that 440Hz is an A4, then it will find an A4. If you tell 440Hz is a C4 it will be a C4. And all the other tones are defined by the temperaments, which are simply tables, which map frequencies to notes. 

Does this help to answer your question?

**Comment by jimishol** [comment]
Yes you answered all my questions clearly.
Thank you very much.

I should pay more attention to [previous discussion](https://github.com/thetwom/Tuner/discussions/69) and do some research on [autocorrelation](https://en.wikipedia.org/wiki/Autocorrelation) that lead me to [Missing fundamental](https://en.wikipedia.org/wiki/Missing_fundamental). The first says that _"The analysis of autocorrelation is a mathematical tool for finding repeating patterns,"_ and the second that _"the general population can be divided into those who perceive missing fundamentals, and those who primarily hear the overtones instead"_. So, these and your post answer perfectly my question.
For my initial post I think that one can accept, as axiom without proof, that the sum of waves will create a pattern that will repeat with a period equal to the least common multiplier of the periods of those individual waves. This is equivalent and equals to the greatest common divisor of the frequencies. (maxima CAS has a bug for the moment and cannot evaluate the gcm of a list, so I am forced to use lcm instead.) I edited and deleted the first of my functions in previous post as the G() one seems to me more correct and consistent. 

As for my observation that "tuner" app gives different pitches when I edit the Temperament option, I WAS WRONG. Autocorrelation in your app works perfectly and gives the expected results for me. In my experiments it gives the 1/lcm or equivalently the  mentioned before missing fundamental. What was wrong was the way i used my phone. The wrong results in some cases were produced because my phone was touching my laptop that produced the testing sound. If one holds the phone with his hands on air, without letting it touch the source of sound, then the results seem perfectly correct. (That observation for how to use the phone might be useful to some READ.ME file.)

Thank you again and I wish you the best!

**Comment by thetwom** [comment]
Glad, that I could help. 

One word about:
>  that the sum of waves will create a pattern that will repeat with a period equal to the least common multiplier of the periods of those individual waves

In an ideal world this is true. Real instrument normally do not have exact harmonics. E.g. assume the fundamental frequency is 440Hz, then the second harmonic ideally would be 880Hz, but on real instruments this will be a bit off, e.g. 881Hz. This means, that you will not really find an exact common multiplier and with this the pattern does not perfectly repeat.
